### PR TITLE
Update RedirectException to extend HttpException.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/Http/Exception/RedirectException.php [hunk 1]
```diff
@@ -31,15 +29,8 @@
  * Additional headers can also be provided in the constructor, or
  * using the addHeaders() method.
```
Comment (othercorey) [inline src/Http/Exception/RedirectException.php:30]: Not sure if we should update or just remove this and let people use the HttpException api naturally.

**Comment by othercorey** [comment]
We need to deprecate the add/remove headers functions.

File: src/Http/Exception/RedirectException.php (hunk 1 above)
Comment (ADmad) [reply src/Http/Exception/RedirectException.php:30]: We could deprecated `addHeaders()`. I am not quite sure how common it is to add new headers after the exception instance has been constructed.

**Comment by othercorey** [comment]
Seems ok. I don't know if setting the http code is necessarily since redirects are handled separately.

File: src/Http/Exception/RedirectException.php [hunk 2]
```diff
@@ -74,21 +69,12 @@ public function addHeaders(array $headers)
      *
      * @param string $key The header to remove.
      * @return $this
+     * @deprecated 4.2.0 Use `setHeaders()` instead.
```
Comment (markstory) [inline src/Http/Exception/RedirectException.php:72]: These methods should emit runtime warnings so we don't have to go back and add them later.

**Review by othercorey — APPROVED** [review]

**Pull request merged**